src/compute-client/src/controller.rs

```diff
@@ -145,6 +145,12 @@ pub struct ComputeController<T> {
     build_info: &'static BuildInfo,
     /// Set to `true` once `initialization_complete` has been called.
     initialized: bool,
+    /// Whether or not this controller is in read-only mode.
+    ///
+    /// When in read-only mode, neither this controller nor the instances
+    /// controlled by it are allowed to affect changes to external systems
+    /// (largely persist).
+    read_only: bool,
```

Comment:
I wonder if we should flip this around and call it `allow_writes` here and everywhere in the protocol. It's not a big deal at all, but `allow_writes` might be nicer for two reasons:

* It matches the `AllowWrites` command, making the the relationship between the two very clear
* It matches how `initialized`, the only other global protocol state variable, is initialized as `false` and then flips to `true`

Reply:
I agree, we have two different terms to describe the same thing, i.e., `read_only` and `allow_writes`. It would be good to unify them to avoid confusion of our future selves.